packages/reactivity/src/effectScope.ts

```diff
@@ -17,8 +17,8 @@ export class EffectScope {
   private index: number | undefined
 
   constructor(detached = false) {
+    this.parent = activeEffectScope
```

Comment:
This is not the correct fix: a detached scope can be turned off at any moment, they don't have a parent because of that: they could be stopped after the active effect scope they were created at has been stopped.

The actual problem is instead in the run function below.

Maybe the `run()` function could just save the activeScope before calling `fn()`:

```ts
  run<T>(fn: () => T): T | undefined {
    if (this.active) {
      const currentEffectScope = activeEffectScope
      try {
        activeEffectScope = this
        return fn()
      } finally {
        activeEffectScope = currentEffectScope
      }
    } else if (__DEV__) {
      warn(`cannot run an inactive effect scope.`)
    }
  }
```

But I'm not completely sure if it should always be `currentEffectScope` or if it should still resolve to the parent (like it does right now) when the effect scope is not detached.

Reply:
It seems to me that clear nest scope is not through this.parent but this.scope, so that's why I assign activeEffectScope to this.parent.

I want to make sure that defination of this.parent is only assigned with undetached scope, Is this right?

Reply:
> I want to make sure the definition of this.parent is only assigned by undetached scope, Is this right?

Yes, a detached scope doesn't have a parent scope because it is detached from where it is created.

Reply:
I refactored the code to keep it simpler as it seems to be enough.